infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -228,22 +244,21 @@ public void prefixTrim(long address) {
         // This is due to the fact that updates on the local datastore don't
         // expose disk sync functionality.
         long newStartingAddress = address + 1;
-        serverContext.setStartingAddress(newStartingAddress);
-        startingAddress = newStartingAddress;
+        dataStore.saveStartingAddress(newStartingAddress);
         syncTailSegment(address);
         log.debug("Trimmed prefix, new starting address {}", newStartingAddress);
     }
 
     private boolean isTrimmed(long address) {
-        return address < startingAddress;
+        return address < dataStore.getStartingAddress();
     }
 
     private void verifyLogs() {
         String[] extension = {"log"};
-        File dir = new File(logDir);
+        File dir = logDir.toFile();
 
         if (!dir.exists()) {
-            return;
+            throw new UnrecoverableCorfuError("Stream log data directory doesn't exists");
```

Comment:
This is changing the behavior of verify. why?

Reply:
Because previously we could go to an invalid state. If the log directory doesn't exist corfu can't start, in the previous version it could happen.